src/components/data-table.tsx

```diff
@@ -85,30 +86,50 @@ const DataTableHead = <Datum extends BasicDatum>({
 );
 const MemoizedDataTableHead = memo(DataTableHead) as typeof DataTableHead;
 
-// Either the rows are selectable
-type Selectable = {
-  /**
-   * A callback function that is called whenever a user selects a row.
-   */
-  onSelectRow: (id: string) => void;
-  /**
-   * An object which indicates which rows have been selected by the user.
-   */
-  selected: string[];
-};
-// Or they're not selectable
-type NonSelectable = {
-  onSelectRow?: never;
-  selected?: never;
+type CellProp<Datum extends BasicDatum> = {
+  column: CommonColumn<Datum>;
+  datum: Datum;
+  loading?: boolean;
+  fixedLayout?: boolean;
+  firstColumn: boolean;
 };
 
-type MaybeSelectable = Selectable | NonSelectable;
+const Cell = <Datum extends BasicDatum>({
+  column,
+  datum,
+  loading,
+  fixedLayout,
+  firstColumn,
+}: CellProp<Datum>) => {
+  let rendered: ReactNode;
+  try {
+    rendered = column.render(datum);
+  } catch (error) {
+    /**
+     * We get here only if the renderer fails. If the renderer returns null of
+     * undefined because of a lack of data, then it will no throw and will not
+     * display the loader at all
+     */
+    if (!loading) {
+      throw error;
+    } else {
+      rendered = firstColumn && <Loader />;
```

Comment:
I'm not sure about this approach. There will be cases where a few columns might be invalid and it doesn't look great. Maybe we could check with UX if there could be a less "in your face" loading indicator we could use here. In the meantime we can keep this as a reminder something should be done about it, or return null, not fussed.

![image](https://user-images.githubusercontent.com/3329765/115201876-b20fb800-a0ed-11eb-8c80-04f3b2217aef.png)

Reply:
Should I propose an alternative visual now, or do you want to get that published asap to be able to use it in your current changes?

Reply:
Let's have that discussion on #ux but you can publish now. We currently have 2 different levels of loading (Loader and the spinner) but it's not really well defined on Franklin what should be used there so this could be done at the same time as adding a column/table (card too?) loading indicator.

Reply:
I've got it linked locally BTW so not in an immediate rush

Reply:
published